### ISSUE TITLE
How to deploy?

### ISSUE THREAD

**Issue by louisabraham** [iodide-project/pyodide#434]
Hi, I made an app that works when using https://iodide-project.github.io/pyodide-demo/pyodide.js and serving from my file system.

However, I cannot get it to work when deploying from a server.

When I use the same URL:

    Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://iodide-project.github.io/pyodide-demo/pyodide.asm.wasm. (Reason: CORS request did not succeed).

I'm currently trying to add every file it asks for, which takes a long amount of time since I'm uploading them to a github pages…

Are there instructions somewhere?

Maybe it would be easier if the CORS policy were different on pyodide-demo?

**Comment by louisabraham** [comment]
I finally made it work with this Makefile: https://github.com/louisabraham/LaBot/blob/master/docs/Makefile

The issue remains as it should be easier.

**Comment by PythonLinks** [comment]
Same situation. 
When I use the pyodide-demo files all works reasonbly well. 
When I try to use the newest release of pyodide, trouble. 
(index):1 Mixed Content: The page at 'https://dev.pythonlinks.info/training/pyodide-training-materials/pyodide-only-web-page/' was loaded over HTTPS, but requested an insecure resource 'http://alpha.iodide.io/pyodide-demo/pyodide.asm.wasm'. This request has been blocked; the content must be served over HTTPS.
(index):1 Uncaught (in promise) TypeError: Failed to fetch
(index):1 Mixed Content: The page at 'https://dev.pythonlinks.info/training/pyodide-training-materials/pyodide-only-web-page/' was loaded over HTTPS, but requested an insecure script 'http://alpha.iodide.io/pyodide-demo/pyodide.asm.data.js'. This request has been blocked; the content must be served over HTTPS.
/favicon.ico:1 Failed to load resource: the server responded with a status of 502 (Bad Gateway)

Basically I want it to get all the files from the same place, but instead it is getting them from someplace else. 

This is a big problem for end users, who just want a simple install of the current v ersion.

**Comment by louisabraham** [comment]
@PythonLinks note that favicon.ico is not needed. I think the best for now is to download the files with my Makefile, it works well for me.

**Comment by mdboom** [comment]
Yeah -- I'll look into how we can change our CORS headers on the Heroku instance running alpha.iodide.io.  That might help.

**Comment by mdboom** [comment]
> @PythonLinks note that favicon.ico is not needed. I think the best for now is to download the files with my Makefile, it works well for me (but non stdlib modules won't work if you don't download them too).

I'd also suggest grabbing these from the releases at https://github.com/iodide-project/pyodide/releases  It was never really intended for pulling from `alpha.iodide.io`.

**Comment by louisabraham** [comment]
@mdboom Thanks for this link, maybe you could mention it in the README? Releases are often not updated so if you intend to build them regularly, it would be better to say it.
I'll update my Makefile to use it instead.

148 MB (270 MB uncompressed) is quite heavy when I need only ~20 MB. Do you think you could provide a stdlib-only version? Scipy in particular is so heavy! 

 A cdn would be a really interesting option to always serve the latest version.

**Comment by PythonLinks** [comment]
A server with CORS headers would be awesome. 
I do not think that my users will  ever add a big load to your server, but it would certainly make life easier for me.  I would not have to download the software, I would not have to configure nginx, I would not have to build a first docker container with the software.  I would not have to upgrade all of that stuff. 

And of course it is not just me, it is all of your end usrs. 

Thank you. 

And check out this reddit.  
https://www.reddit.com/r/Python/comments/br9wsj/python_webapp_where_python_runs_on_within_the/

A brilliant use case of Pyodide.  For medical records, so that the data never leaves the hospital.  

My daughter has the reverse problem, she has to log into the server to access the data.

**Comment by phorward** [comment]
> 148 MB (270 MB uncompressed) is quite heavy when I need only ~20 MB. Do you think you could provide a stdlib-only version? Scipy in particular is so heavy!

That would be perfect!

**Comment by mdboom** [comment]
If you don't use the extra packages, they are never downloaded and use no resources.  You can safely delete them from your own deployment if you are deploying yourself.

**Comment by louisabraham** [comment]
Hi, is there a plan to use a CDN at some point? It would probably be the best option.

**Comment by mdboom** [comment]
Pyodide is already deployed to a non-versioned CDN [see here](https://github.com/iodide-project/pyodide/blob/master/docs/using_pyodide_from_javascript.md#startup).

Eventually we'd like to have a versioned one, but we exceed the size limits of most free hosting.

**Comment by louisabraham** [comment]
I guess this solves the issue.

Thank you for your amazing work!
